controllers/openshift_controller.go

```diff
@@ -1679,6 +1679,69 @@ func (r *KataConfigOpenShiftReconciler) getMcpByName(mcpName string) (*mcfgv1.Ma
 	return mcp, nil
 }
 
+func (r *KataConfigOpenShiftReconciler) processTransitioningNode(node *corev1.Node) error {
+	// Don't bother checking the second return value, the annotation is there
+	// otherwise we wouldn't be here (our caller checked already).
+	mcoNodeState, _ := node.Annotations["machineconfiguration.openshift.io/state"]
+
```

Comment:
I actually don't agree with treating every single function like it was part of a public API, callable by anyone.  This is a private function, a helper to `updateStatus()` specifically.  It's only reason to be is that the code is too bulky to fit into a `case` arm of the `switch` statement in its only caller so I factored it out - it's not a function that anybody else will ever call.

So, because `updateStatus()` and `processTransitioningNode()` (and `processDoneNode()` too) form a tight subsystem, I believe it's appropriate for the helper to rely on work done by the caller.

Reply:
We don't agree then. Since we don't know who will have to maintain this code in the long term, we should do best efforts to have clear APIs, even internally. Putting all the code in the `case` statement of `updateStatusNew()` is certainly not what I'm suggesting.

I've approved the PR anyway :wink: